sky/backends/cloud_vm_ray_backend.py

```diff
@@ -624,13 +625,15 @@ def __init__(
             cluster_name: str,
             resources: resources_lib.Resources,
             num_nodes: int,
-            prev_cluster_status: Optional[status_lib.ClusterStatus],
-            prev_handle: Optional['CloudVmRayResourceHandle'],
+            ports_to_open: Optional[List[str]] = None,
```

Comment:
The `ports` section inside `resources` represents the ports after this launch, since we need to write the newest version to cluster config (provider section) through the `backend_utils.write_cluster_config` function... Here the `ports_to_open` is just the new ports that will be opened in this `sky launch`

Reply:
nvm, I just noticed that by changing rule names for GCP into `user-ports-{cluster_name}-{incremental_serial_number}` we could completely get rid of the ports section in the provider config (since we don't need to keep track of firewall rules by ports, but just a prefix); and we could just ignore the `ports` section in `to_provision` and as you said below, just check the previous handle with the current handle to get the ports to open now 🫡